nexus/src/authz/actor.rs

```diff
@@ -0,0 +1,93 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Oso integration for Actor types
+
+use super::roles::RoleSet;
+use crate::authn;
+use omicron_common::api::external::ResourceType;
+use uuid::Uuid;
+
+/// Represents [`authn::Context`] (which is either an authenticated or
+/// unauthenticated actor) for Polar
+#[derive(Clone, Debug)]
+pub struct AnyActor {
+    authenticated: bool,
+    actor_id: Option<Uuid>,
+    roles: RoleSet,
+}
+
+impl AnyActor {
+    pub fn new(authn: &authn::Context, roles: RoleSet) -> Self {
+        let actor = authn.actor();
+        AnyActor {
+            authenticated: actor.is_some(),
+            actor_id: actor.map(|a| a.0),
+            roles,
+        }
+    }
+}
+
+impl PartialEq for AnyActor {
+    fn eq(&self, other: &Self) -> bool {
+        self.actor_id == other.actor_id
+    }
+}
```

Comment:
When do `AnyActor`s get compared? Is there a risk of creating two actors with the same ID but different `RoleSets` and considering them equal when they're not supposed to be?

Reply:
tl;dr: I think this implementation is vestigial.  I'm removing it in 2ffc5b90.

In theory this _could_ be used in rules in the omicron.polar policy file, like "this actor is allowed to access this resource if they are the built-in 'db-init' user" or "this actor is allowed to access this resource if they are the owner".  I think we'd only ever do this with `AuthenticatedActor` (and we _do_ do that today [here](https://github.com/oxidecomputer/omicron/pull/520/files/9d12b9e7354691aac0139b6b2d7bd06b6742cd85#diff-79878c4356f0c780fbde8574a057daf8129f2205f3a1165d2d6ce5d42e8b148dR44)).

Reply:
Also: in the case that we're comparing two `AuthenticatedActor`s, it doesn't matter what their `RoleSet` is -- we're trying to determine if they're the same person (or machine, or whatever).